src/iexamine.cpp

```diff
@@ -1774,88 +1830,225 @@ void iexamine::aggie_plant(player &p, const tripoint &examp)
             } else if( (p.has_trait(trait_M_DEFENDER)) || ( (p.has_trait(trait_M_SPORES) || p.has_trait(trait_M_FERTILE)) &&
                 one_in(2)) ) {
                 g->summon_mon( mon_fungal_blossom, examp );
-                add_msg(m_info, _("The seed blooms forth!  We have brought true beauty to this world."));
-            } else if ( (p.has_trait(trait_THRESH_MYCUS)) || one_in(4)) {
-                g->m.furn_set(examp, f_flower_marloss);
-                add_msg(m_info, _("The seed blossoms rather rapidly..."));
+                add_msg( m_info, _( "The seed blooms forth!  We have brought true beauty to this world." ) );
+            } else if( ( p.has_trait( trait_THRESH_MYCUS ) ) || one_in( 4 ) ) {
+                g->m.furn_set( examp, f_flower_marloss );
+                add_msg( m_info, _( "The seed blossoms rather rapidly..." ) );
             } else {
-                g->m.furn_set(examp, f_flower_fungal);
-                add_msg(m_info, _("The seed blossoms into a flower-looking fungus."));
+                g->m.furn_set( examp, f_flower_fungal );
+                add_msg( m_info, _( "The seed blossoms into a flower-looking fungus." ) );
             }
-        } else { // Generic seed, use the seed item data
-            const itype &type = *seed.type;
-            g->m.i_clear(examp);
-            g->m.furn_set(examp, f_null);
-
-            int skillLevel = p.get_skill_level( skill_survival );
-            ///\EFFECT_SURVIVAL increases number of plants harvested from a seed
-            int plantCount = rng(skillLevel / 2, skillLevel);
-            if (plantCount >= 12) {
-                plantCount = 12;
-            } else if( plantCount <= 0 ) {
-                plantCount = 1;
-            }
-            const int seedCount = std::max( 1l, rng( plantCount / 4, plantCount / 2 ) );
-            for( auto &i : get_harvest_items( type, plantCount, seedCount, true ) ) {
-                g->m.add_item_or_charges( examp, i );
+        } else {
+            std::list<item> harvest_items = get_harvest_items( seed );
+            proceed_plant_after_harvest( examp );
+            for( auto &i : harvest_items ) {
```

Comment:
After some debugging I found what the crash can occur at this block, where new items are added to the map. It occurs only if new perishable items are added (cannot reproduce only with withered plants). Not sure if it is related to new changes, because reloading the game and doing literally the same actions will produce the same list of items (even with the same bday time) and it works fine almost every time.